src/ILCompiler.Compiler/src/Compiler/Compilation.cs

```diff
@@ -21,11 +21,23 @@ namespace ILCompiler
 {
     public class CompilationOptions
     {
-        public IReadOnlyDictionary<string, string> InputFilePaths;
-        public IReadOnlyDictionary<string, string> ReferenceFilePaths;
+        /// <summary>
+        /// Associate for each simple name, the list of assembly full path names matching the simple name.
+        /// Useful when including multiple assemblies with same name from different locations.
```

Comment:
It does make sense but not making that check can yield surprising results as shown in my test case (albeit a quite specific rare scenario but supported by the .NET specification). One of the reasons I made this is because we have been having a lot of issues with aot on various platforms with the work we do and detecting issues early on is better.